test/extended/openstack/bug_1765.go

```diff
@@ -0,0 +1,155 @@
+			newMachinesetParams := framework.BuildMachineSetParams(ctx, rclient, 1)
+			rawBytes, err = json.Marshal(newMachinesetParams.ProviderSpec.Value)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error marshaling new MachineSet Provider Spec")
+			err = json.Unmarshal(rawBytes, &newProviderSpec)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error unmarshaling new MachineSet Provider Spec")
+			newMachinesetParams.Name = fmt.Sprintf("%v-%v", "extra-network", RandomSuffix())
+			newProviderSpecJson, err := json.Marshal(newProviderSpec)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Failed to marshal new Machineset provider spec")
+			newMachinesetParams.ProviderSpec.Value.Raw = newProviderSpecJson
+
+			// Set NoAlloedAddressPairs for the additional network to true
+			var extraNetworkParam machinev1alpha1.NetworkParam
+			extraNetworkParam.NoAllowedAddressPairs = true
+			extraNetworkParam.Filter.ID = extraNetwork.ID
+			newProviderSpec.Networks = append(newProviderSpec.Networks, extraNetworkParam)
+
+			g.By("Create a new machineSet with an additional network with NoAllowedAddressPairs set to true")
+			ms, err := framework.CreateMachineSet(rclient, newMachinesetParams)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Failed to create a Machineset")
+			defer DeleteMachinesetsDefer(rclient, ms)
+			err = GetMachinesetRetry(ctx, rclient, ms, true)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Failed to get the new Machineset")
+			err = waitUntilNMachinesPrefix(ctx, dc, ms.Name, 1)
+
+			newMachines, err := machines.List(ctx, dc, machines.ByMachineSet(ms.Labels["machine.openshift.io/cluster-api-machineset"]))
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error fetching machines for MachineSet %q", ms.Name)
+
+			var machineNetworks []string
+			for _, machine := range newMachines {
+				for _, net := range objects(machine.Get("spec.providerSpec.value.networks")) {
+					machineNetworks = append(machineNetworks, net.Get("filter.id").String())
+				}
+			}
+			g.By("Verify ports AllowedAddressPairs")
+			for _, net := range machineNetworks {
+				portListOpts := ports.ListOpts{
+					NetworkID: net,
+				}
+				allPages, err := ports.List(networkClient, portListOpts).AllPages(ctx)
+				o.Expect(err).NotTo(o.HaveOccurred(), "Failed to get ports")
+				allPorts, err := ports.ExtractPorts(allPages)
+				o.Expect(err).NotTo(o.HaveOccurred(), "Failed to extract ports")
+				for _, port := range allPorts {
+					if strings.Contains(port.Name, newMachinesetParams.Name) {
+						if net == extraNetwork.ID {
+							// There should be a port in the additional network with
+							// an empty AllowedAddressPairs
+							o.Expect(len(port.AllowedAddressPairs)).To(o.Equal(0))
+						} else {
+							// For other ports AllowedAddressPairs shoudn't be empty
+							o.Expect(len(port.AllowedAddressPairs)).To(o.Not(o.Equal(0)))
+						}
+					}
+				}
+			}
+			time.Sleep(10 * time.Second)
+			g.By("Deleting the new machineset")
```

Comment:
This should be on an AfterEach. If not, the machineset won't be deleted if the TC fails.

Reply:
oh! ok :)